### PR TITLE
Better validation errors

### LINKED ISSUE #74: Validation errors don't always snakify field names correctly
E.g. I keep seeing `start_flow: field 'uui_d' required`

### PR DESCRIPTION
**Pull request by rowanseymour**
For example: 

```json
"errors": [
    "field 'created_on' on 'event[type=msg_received]' is required"
]
```

Fixes #74

### CODE REVIEW

File: cmd/docgen/actions.go [hunk 1]
```diff
@@ -40,7 +40,7 @@ func handleActionDoc(output *bytes.Buffer, prefix string, typeName string, docSt
 	}
 
 	// validate it
-	err = utils.ValidateAll(action)
```
Comment (rowanseymour) [inline cmd/docgen/actions.go]: We never use this to validate multiple objects

File: cmd/flowserver/main.go [hunk 2]
```diff
@@ -114,7 +114,7 @@ func writeError(w http.ResponseWriter, r *http.Request, status int, err error) e
 	lg.Log(r.Context()).WithError(err).Error()
 	errors := []string{err.Error()}
 
-	vErrs, isValidation := err.(utils.ValidationError)
+	vErrs, isValidation := err.(utils.ValidationErrors)
```
Comment (rowanseymour) [inline cmd/flowserver/main.go]: This is just an array of errors now so this name seems apt

File: flows/actions/envelope.go [hunk 3]
```diff
@@ -1,77 +1,43 @@
...
-		action := WebhookAction{}
-		err := json.Unmarshal(envelope.Data, &action)
-		return &action, utils.ValidateAllUnlessErr(err, &action)
-
+		action = &WebhookAction{}
 	default:
 		return nil, fmt.Errorf("Unknown action type: %s", envelope.Type)
 	}
+
+	return action, utils.UnmarshalAndValidate(envelope.Data, action, fmt.Sprintf("action[type=%s]", envelope.Type))
```
Comment (rowanseymour) [inline flows/actions/envelope.go]: This is the pattern we used most and should make as easy as possible - we unmarshal into an envelope and need to validate the envelope.

File: flows/engine/assets.go [hunk 4]
```diff
@@ -54,7 +55,8 @@ type assetsEnvelope struct {
 
 func ReadAssets(data json.RawMessage) (flows.Assets, error) {
 	var envelope assetsEnvelope
-	if err := json.Unmarshal(data, &envelope); err != nil {
+	err := utils.UnmarshalAndValidate(data, &envelope, "assets")
+	if err != nil {
```
Comment (nicpottier) [inline flows/engine/assets.go]: The former is more go-y, but I actually haven't gotten comfortable with it yet..

**Comment by rowanseymour** [comment]
@nicpottier have done "structural validation" in different PR: https://github.com/nyaruka/goflow/pull/78

**Pull request merged**